karate-core/src/main/java/com/intuit/karate/core/ScenarioRuntime.java

```diff
@@ -56,12 +59,21 @@
     public final Map<String, Object> magicVariables;
     public final boolean selectedForExecution;
     public final boolean dryRun;
+    public final boolean isBackgroundRuntime;
 
     public ScenarioRuntime(FeatureRuntime featureRuntime, Scenario scenario) {
-        this(featureRuntime, scenario, null);
+        this(featureRuntime, scenario, null, false);
+    }
+
+    public ScenarioRuntime(FeatureRuntime featureRuntime, Scenario scenario, boolean backgroundRuntime) {
```

Comment:
do we need the extra boolean won't it be a property of the `ScenarioRuntime`

Reply:
When running Scenario Outlines the Background runs wrapped in a ScenarioRuntime - while when running a scenario the background section is just included in the steps. That background ScenarioRuntime doesn't have a scenario associated with thus I had a nullpointer.

I think there was a way to figure out (like null scenario maybe or something) that I can use instead of the boolean to avoid evaluating the scenario name.

Reply:
try `scenario.isDynamic()` you can see it used elsewhere in the same file